### PR TITLE
Upgrade all dependencies for 3.0.0 release with support for Python 3.8 - 3.12

### PR DESCRIPTION
**Pull request by somexpert**

### CODE REVIEW

**Comment by somexpert** [comment]
ping @ldko, @clarktr1, this is ready for review. This upgrades the app to be compatible with Python versions 3.8 - 3.12, dropping support for previous versions. The README provides general details on setting this up locally for testing, but I'm also happy to help fill in any gaps.

**Comment by somexpert** [comment]
The larger test changes are due to Flask getting more particular on how they allow access to the ["application context"](https://flask.palletsprojects.com/en/3.0.x/appcontext/#the-application-context), and so I had to change things to use their [newer preferred method](https://flask.palletsprojects.com/en/2.3.x/testing/#tests-that-depend-on-an-active-context) for passing in a context instead of relying on a mock.

File: README.md [hunk 1]
```diff
@@ -74,45 +74,34 @@ directions to start developing/testing.
 
    Please see "default_settings.py" for an example of how a settings file should look.
 
-4. Set the following two environment variables so that Flask knows which app to run and
-   to do so in the "development" mode.
+4. Start the app.
     ```sh
-        $ export FLASK_APP=aubrey_transcription
-        $ export FLASK_ENV=development
+        $ flask --app aubrey-transcription run --debug
```
Comment (ldko) [inline README.md]: As-is for me, this command results in `Error: Could not import 'aubrey-transcription'.` Swap dash for underscore.
```suggestion
        $ flask --app aubrey_transcription run --debug
```

**Review by ldko — APPROVED** [review]
Works for me, just needs a README typo fix to run the app. :+1: .

**Comment by clarktr1** [comment]
36/36 passed for me 👍

**Review by clarktr1 — APPROVED** [review]
36/36 passed for me 👍

**Pull request merged**